### ISSUE TITLE
"Image cannot be loaded" for valid GIFs

### ISSUE THREAD

**Issue by iamcarbon** [SixLabors/ImageSharp#405]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] Only checked in Release
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

These two GIFS are throwing "NotSupportedException | Image cannot be loaded." with a list of the availability decoders:

https://carbonmade-media.accelerator.net/34349616
https://carbonmade-media.accelerator.net/19459558

I've verified that both these image's headers start with "GIF89a", so it looks like we may be throwing after decoder selection deeper down the stack. They both load correctly in WIC, Skia, and OSX.

I'll clone master and see if I can get us some more details later today. At the very least, it would be helpful to specialize the exception here.

- ImageSharp version: 1.0.0-beta2

**Comment by JimBobSquarePants** [comment]
@iamcarbon I've had a quick look and have identified the issue.

Those two images have Application Extension Blocks with an incorrect length of 252. I checked the spec and it should be 11.

Here's a great description.
http://www.matthewflickinger.com/lab/whatsinagif/bits_and_bytes.asp#application_extension_block

Anyway, it's an easy fix as the header actually tells us what length to skip so I've sorted that locally.

I'll have a quick look at #403 also and see if I can push a fix for both at the same time.
